src/ReplaySession.cc

```diff
@@ -273,6 +276,13 @@ Completion ReplaySession::cont_syscall_boundary(
   return COMPLETE;
 }
 
+void ReplaySession::clear_syscall_bp() {
+  if (syscall_bp_vm) {
+    syscall_bp_vm->remove_breakpoint(syscall_bp_addr, BKPT_INTERNAL);
+    syscall_bp_vm = nullptr;
```

Comment:
Seems to me that if all tasks for this AddressSpace die before this gets called, we could have an AddressSpace with no Tasks, in which case remove_breakpoint will crash because we don't find a usable Task to reset the memory through. Need to modify the AddressSpace code so that if there's no Task we don't try to do anything (since the memory is already gone in fact).